docs/guides/server/caching.adoc

```diff
@@ -108,21 +108,42 @@ CPU, memory, and network utilization.
 
 These in-memory caches for user sessions and client sessions are limited to, by default, 10000 entries per node which reduces the overall memory usage of {project_name} for larger installations.
 The internal caches will run with only a single owner for each cache entry.
-Consider trade-off between memory consumption and the database utilization and set different sizes for the caches, edit {project_name}'s cache config file (`conf/cache-ispn.xml`) to set a `+<memory max-count="..."/>+` for those caches.
 
-.Volatile user sessions
+.Offline user sessions
+As an OpenID Connect Provider, the server is also capable of authenticating users and issuing offline tokens. Similarly to regular user and client sessions,
+when an offline token is issued by the server upon successful authentication, the server also creates an offline user session and an offline client session.
 
-By default, user sessions are stored in the database and loaded on-demand to the cache.
-It is possible to configure {project_name} to store user sessions in the cache only and minimize the database utilization.
+The following caches are used to store offline sessions:
+
+* offlineSessions
+* offlineClientSessions
+
+Similarly to regular user and client sessions caches, also the offline user and client session caches are limited to 10000 entries per node by default. Items which are evicted from the memory will be loaded on-demand from the database when needed.
+
+.Password brute force detection
+The `loginFailures` distributed cache is used to track data about failed login attempts.
+This cache is needed for the Brute Force Protection feature to work in a multi-node {project_name} setup.
+
+.Action tokens
+Action tokens are used for scenarios when a user needs to confirm an action asynchronously, for example in the emails sent by the forgot password flow.
+The `actionTokens` distributed cache is used to track metadata about action tokens.
+
+=== Volatile user sessions
+
+By default, regular user sessions are stored in the database and loaded on-demand to the cache.
+It is possible to configure {project_name} to store regular user sessions in the cache only and minimize the database utilization.
 
 Since all the sessions in this setup are stored in-memory, there are two side effects related to this:
 
-* Losing sessions on when all {project_name} nodes restart.
+* Losing sessions when all {project_name} nodes restart.
 * Increased memory consumption.
 
+Using volatile user sessions also disables an improved handling of offline sessions, where only a limited number of offline sessions needs to be loaded into memory.
+Due to this, apply the settings below not only to the regular session caches but also the offline session caches.
```

Comment:
We're saying the same thing multiple times here. I think we can make this section clearer by stating that the cache cannot have an upper bound here and then in the steps just say what needs to be changed. Something like:

"When using volatile user sessions, the cache is the source of truth for user and client sessions, therefore it is necessary to not define an upper limit on the total number of entries that can be stored. Similarly, it's crucial that we replicate each entry to at least two nodes to prevent data loss on node failures and rolling restarts.

Follow these steps to enable this setup:

1. Edit {project_name}'s cache config file (`conf/cache-ispn.xml`) for caches `sessions`, `clientSessions`, `offlineSessions` and `offlineClientSessions` with the following update:

"

Reply:
There is the unfortunate interconnection that people need to change the regular and offline cache configuration. So I would like to keep that one paragraph, but moved it a bit higher. Please have a look.